tests/suite/cli_self_upd.rs

```diff
@@ -350,13 +350,12 @@ async fn update_overwrites_programs_display_version() {
         .await;
 
     USER_RUSTUP_VERSION
-        .set_value(Some(PLACEHOLDER_VERSION))
+        .set_value(Some(PLACEHOLDER_VERSION.as_bytes()))
         .unwrap();
     cx.config.expect_ok(&["rustup", "self", "update"]).await;
-    assert_eq!(
-        USER_RUSTUP_VERSION.get_value::<String>().unwrap().unwrap(),
-        version,
-    );
+    assert_eq!(USER_RUSTUP_VERSION.get_value().unwrap().unwrap(), {
+        version.as_bytes()
+    });
```

Comment:
I don't know which side should I modify. CI failure pointed out that the left side is probably a UTF-16 string (seems also have a terminated `\0`?) which was represented with an `u8` array, while the right side is a pure UTF-8 string.